models/user/userModel.js

```diff
@@ -0,0 +1,66 @@
+import mongoose, { Schema } from 'mongoose';
+import { addressSchema, orderItemSchema } from '../schemas';
+import { accountStatuses, roles } from '../../constants';
+import { phoneNumberPattern } from '../../helpers';
+
+
+const ObjectId = Schema.Types.ObjectId;
+
+const userSchema = new Schema(
+  {
+    fullName: {
+      type: String,
+      required: [true, 'Full name is required'],
+      minlength: [3, 'Full name must be at least 3 characters long'],
+    },
+    password: {
+      type: String,
+      required: [true, 'Password is required'],
+      minlength: [8, 'Password must be at least 8 characters long'],
+    },
+    email: {
+      type: String,
+      required: [true, 'Email is required'],
+      unique: true,
+      validate: {
+        validator: (value) => /\S+@\S+\.\S+/.test(value),
+        message: 'Invalid email format',
+      },
+    },
+    avatar: String,
```

Comment:
I saw `imageUrl` or something like that in other schemas, duplicates?

Reply:
For the chef and courier models, my colleague wrote "chefImage" and "courierImage" accordingly. But in Miro, we agreed on a template name for this field for all roles, and it's "avatar"

Reply:
I mean why do we need two fields for user's picture: `avatar` in `User` model and `chefImage` in `Chef` model?

Reply:
For a user, attaching a photo to their account is not as important. Meanwhile, all users can visit the profile of a chef, and they would like to see his professional photo.

So, 
in the user model - the avatar is optional.
in the chef's model - the avatar is mandatory, with the possibility to add a professional photo.
in the courier model - the avatar is mandatory, with the possibility to add another professional photo.

Reply:
It's a bit confusing logic at least I would say, but if there's no other way leave it as it is